### ISSUE TITLE
Investigate AnyM handling of single valued monads versus monads with multiple values

### ISSUE THREAD

**Issue by johnmcclean-aol** [aol/cyclops#75]
a.
```java        
       System.out.println(AnyM.fromOptional(Optional.of(1)).flatMap(i->AnyM.fromSet(new HashSet(Arrays.asList(1,2,3)))));

        //AnyM(MonadWrapper(monad=Optional[[1, 2, 3]], orgType=class java.util.Optional))
       Optional<HashSet<Integer>>
```
b.
```java  
		System.out.println(AnyM.fromOptional(Optional.of(1)).flatMap(i->AnyM.fromStream(Stream.of(1,2,3))));
       //AnyM(MonadWrapper(monad=Optional[[1, 2, 3]], orgType=class java.util.Optional))
       Optional<List<Integer>>
```
c.
```java 
		System.out.println(AnyM.fromOptional(Optional.of(1)).flatMap(i->AnyM.fromStream(Stream.of(1))));
       //AnyM(MonadWrapper(monad=Optional[[1, 2, 3]], orgType=class java.util.Optional))
       Optional<List<Integer>>
```
(c) The last option could be 
```java  
Optional<Integer>
``` 
as it is from a single valued stream.

(b) + (c) could be 
```java 
Optional<Stream<Integer>>
```

(a) + (b) + (c) could result in 
```java 
Optional<Integer>
``` 
taking the first value only.

**Comment by johnmcclean-aol** [comment]
Within AnyM we will take the first value during flatMap operations that result in multiple results for a single value placeholder (i.e. Optional.of(1).flatMap(i->Stream.of(1,2,i)) results in Optional.of(1)

Outside of AnyM (internally and in for-comprehensions)  the behaviour will be 

Optional.of(1).flatMap(i->Stream.of(1,2,i)) results in Optional.of(List(1,2,1))

**Comment by johnmcclean-aol** [comment]
Released in v7.0.0 of cyclops
